external/avro/src/main/scala/org/apache/spark/sql/avro/SchemaConverters.scala

```diff
@@ -51,11 +52,21 @@ object SchemaConverters {
     toSqlTypeHelper(avroSchema, Set.empty)
   }
 
+  // The property specifies Catalyst type of the given field
+  private val CATALYST_TYPE_PROP_NAME = "spark.sql.catalyst.type"
+
   private def toSqlTypeHelper(avroSchema: Schema, existingRecordNames: Set[String]): SchemaType = {
     avroSchema.getType match {
       case INT => avroSchema.getLogicalType match {
         case _: Date => SchemaType(DateType, nullable = false)
-        case _ => SchemaType(IntegerType, nullable = false)
+        case _ =>
+          val catalystTypeAttrValue = avroSchema.getProp(CATALYST_TYPE_PROP_NAME)
+          val catalystType = if (catalystTypeAttrValue == null) {
+            IntegerType
+          } else {
+            CatalystSqlParser.parseDataType(catalystTypeAttrValue)
```

Comment:
Can we add an assert to make sure it's year-month interval? I agree to use a general property `spark.sql.catalyst.type`, which can be useful in the future. But I'm a bit afraid to return an arbitrary type here. It needs to be compatible with a physical avro INT at least.

Reply:
> It needs to be compatible with a physical avro INT at least.

The deserializer should fail at https://github.com/apache/spark/pull/34503/files#diff-f82299271bc612e2032bc9e22f0f2ed3af0eccb2a37acdd6a25cc1d999c1be92R333-R339 with proper error message telling users about the incompatible schema. I am not sure about assert, especially, for **external** inputs. From my point of view, asserts are for checking contracts between **internal** components.

Reply:
ah I see, the deserializer can check it. LGTM!